orchestration/dagster/hca_orchestration/resources/base.py

```diff
@@ -0,0 +1,165 @@
+import subprocess
+from uuid import uuid4
+
+import kubernetes
+from dagster import resource, StringSource, Field
+from dagster_k8s.client import DagsterKubernetesClient
+from google.cloud import storage
+from google.auth.transport.requests import Request
+import google.auth
+from data_repo_client import ApiClient, Configuration, RepositoryApi
+
+ONE_DAY_IN_SECONDS = 86400  # seconds
+POLLING_INTERVAL = 5  # seconds
+
+
+@resource({
+    "working_dir": Field(StringSource)
+})
+def local_beam_runner(init_context):
+    return LocalBeamRunner(working_dir=init_context.resource_config["working_dir"])
+
+
+class LocalBeamRunner:
+    def __init__(self, working_dir):
+        self.working_dir = working_dir
+
+    def run(self, job_name, input_prefix, output_prefix, context):
+        context.log.info("Local beam runner")
+        # TODO this is hardcoded to the HCA transformation pipeline for now
+        subprocess.run(
+            ["sbt", f'hca-transformation-pipeline/run --inputPrefix={input_prefix} --outputPrefix={output_prefix}'],
+            check=True,
+            cwd=f"{self.working_dir}"
+        )
+
+
+@resource({
+    "project": Field(StringSource),
+    "temp_location": Field(StringSource),
+    "subnet_name": Field(StringSource),
+    "service_account": Field(StringSource),
+    "image_name": Field(StringSource),
+    "image_version": Field(StringSource),
+    "namespace": Field(StringSource)
+})
+def dataflow_beam_runner(init_context):
+    return DataflowBeamRunner(
+        project=init_context.resource_config['project'],
+        temp_location=init_context.resource_config['temp_location'],
+        subnet_name=init_context.resource_config['subnet_name'],
+        service_account=init_context.resource_config['service_account'],
+        image_name=init_context.resource_config['image_name'],
+        image_version=init_context.resource_config['image_version'],
+        namespace=init_context.resource_config['namespace']
+    )
+
+
+class DataflowBeamRunner:
+    def __init__(self, project, temp_location, subnet_name, service_account, image_name, image_version, namespace):
```

Comment:
noting for later: there are some python libraries we can pull in to make attribute assignment boilerplate like this a lot more terse